clowder-framework/clowder2
# Annotate image widget

**The widget is working  but for proper testing it would be good to have a sample extractor  for annotating images in pyclowder, I finished the rough draft of it already **

## Aim
The aim of this widget is to annotate images using user points to create a polygon around an object, the code could also be reused for other image based tasks like image prompting

## How to test

In the front end directory run npm install as we have a couple of new libraries 

Run an extractor, locally as in to do the necessary modifications. The extractor should be used on image files. For ease, I created a sample extractor in the pyclowder repo ([branch is called sample-image-annotation-extractor](https://github.com/clowder-framework/pyclowder/tree/sample-image-annotation-extractor/sample-extractors/image-annotation-extractor)). Running the extractor, is the best way to test the widget.

  1. Add another parameter in extractor-info.json file,
  ```json
  "parameters": {
      "schema": {
        "Annotations": {
          "type": "string",
          "title": "Annotate image",
          "description": "Annotate image",
          "format": "ImageAnnotator"
        }
      }
    }
  ```
  2. When you open the extractor, you can see a button that opens a modal 
  ![image](https://github.com/clowder-framework/clowder2/assets/16596733/f3cfe18a-8649-4c2b-a5b6-e45589590254)
  ![image](https://github.com/clowder-framework/clowder2/assets/16596733/6f770665-cfbb-414e-a4c9-42a352b821e4)

Clicking on the image should create points that form a polygon around the desired object, add a name to save annotation.

**PS I also made some changes regarding the FileSelector Component to make the naming consistent**
